### ISSUE TITLE
사용자인증(2) - 장고(Django) 핥짝 맛보기

### ISSUE THREAD

**Issue by swarf00** [swarf00/swarf00.github.io#10]
https://swarf00.github.io/2018/12/10/login.html 

 1. 로그인 기능 생성가입하기 기능을 구현하면서 장고의 모델폼과 템플릿에 대해 많이 익숙해졌습니다. 로그인 기능을 구현할 때는 이보다는 복잡하지 않을 겁니다. 로그인은 어느 상황에서도 할 수 있도록 화면 상단(내비게이션바)의 오른쪽에 보이도록 할 겁니다. 로그인이 된 상태라면 당연...

**Comment by Kim-JinYong** [comment]
LOGIN_URL = '/user/login/' 추가하고,
bbs/views.py 에 from django.conf import settings 해야
login_url = settings.LOGIN_URL 을 사용할 수 있는 것 같습니다!

**Comment by sehunone** [comment]
> @Kim-JinYong
> LOGIN_URL = '/user/login/' 추가하고,
> bbs/views.py 에 from django.conf import settings 해야
> login_url = settings.LOGIN_URL 을 사용할 수 있는 것 같습니다!
> 

import 구문을 빼먹었네요.ㅠㅠ 지적해주셔서 감사해요. 수정해서 올릴께요 ^^

**Comment by Kim-JinYong** [comment]
추가로 
user/views.py 에 from django.contrib import messages
도 하셔야 로그인 시 messages.error 작동 합니다.

**Comment by sehunone** [comment]
> @Kim-JinYong
> 추가로 
> user/views.py 에 from django.contrib import messages
> 도 하셔야 로그인 시 messages.error 작동 합니다.

맞습니다. [Template 만들기](https://swarf00.github.io/2018/11/23/build-template.html#4-%EA%B2%8C%EC%8B%9C%EB%AC%BC-%EC%97%85%EB%8D%B0%EC%9D%B4%ED%8A%B8-%ED%85%9C%ED%94%8C%EB%A6%BF-%EA%B5%AC%ED%98%84) 에서 설명이 되어 있으나 중복되어 생략한 부분인데 못 보신 분들은 모를 수도 있겠네요

**Comment by smilejakdu** [comment]
안녕 하세요 
OperationalError at /user/login/
no such table: user_user

이러한 에러가 발생해서 
model 부분을 봤는데, 
어디를 수정해야할 지 몰라서 댓글을 남겨봅니다.

**Comment by sehunone** [comment]
`no such table` 이란 오류는 model은 정의되어 있으나 마이그레이션이 정상적으로 되지 않아 발생하는 오류입니다.
콘솔에서 `./manage.py migrate` 명령어를 실행 후 다시 서버를 실행하시거나 실제 데이터베이스에 user_user 테이블이 정상적으로 생성되어 있는 지 확인해보세요.

> 안녕 하세요 
> OperationalError at /user/login/
> no such table: user_user
> 
> 이러한 에러가 발생해서 
> model 부분을 봤는데, 
> 어디를 수정해야할 지 몰라서 댓글을 남겨봅니다.

**Comment by KimDoKy** [comment]
장고에서의 세션, 인증에 대해 잘 보고 갑니다 :)

**Comment by swarf00** [comment]
> @KimDoKy
> 장고에서의 세션, 인증에 대해 잘 보고 갑니다 :)

장고 소스를 살펴보면 그다지 어려운 것이 없는데 선뜻 열어보기 두렵죠. ㅠㅠ
장고 개발자들이 지들 입으로 완벽주의자들이기 때문에 장고는 꽤 완성도가 높다고 했는데, 실제로 소스코드를 보면 많은 부분 동의가 됩니다.
장고 뿐만 아니라 파이썬에 대해서 좀 더 깊이 공부를 해보고 싶으시면 장고 소스코드를 열고 공부해보시길 바랍니다.

**Comment by KimDoKy** [comment]
엇. 대댓글이 아니라 메일을 주셨네요 ㅎ
세션, 인증 관련해서는 가장 잘 설명해 주신 것 같습니다. 덕분에 이해에 큰 도움이 되었습니다.
기능들을 파고들다보면 자연스럽게 장고뿐만 아니라 서드파티 소스파일까지 다 까보게 되는 것 같습니다 :)

2020년 3월 5일 (목) 오후 1:16, swarf00 <notifications@github.com>님이 작성:

**Comment by id8436** [comment]
와;; 도움되는 글 정말 감사합니다. 혹시 권한그룹을 따로 지정하는 방법은 없을까요? 특정한 권한이 없으면 보이지 않게 한다든가..
